src/Execution/Arguments/ArgumentSet.php

```diff
@@ -77,12 +77,20 @@ function (Directive $directive): bool {
     /**
      * Apply ArgBuilderDirectives and scopes to the builder.
      *
-     * @param  \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder  $builder
-     * @param  string[]  $scopes
+     * @param \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder $builder
+     * @param string[] $scopes
+     * @param \Closure $directiveFilter
+     *
      * @return \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder
      */
-    public function enhanceBuilder($builder, array $scopes)
+    public function enhanceBuilder($builder, array $scopes, $directiveFilter = null)
     {
+        if (empty($directiveFilter)) {
+            $directiveFilter = function (Directive $directive): bool {
```

Comment:
I think we should apply the extra filter in addition to filtering for `ArgBuilderDirective`, because the function afterwards can only continue to work with such instances.